components/BlogSectionHeader/index.tsx

```diff
@@ -0,0 +1,44 @@
+import { IPost } from "components/Post";
+import PostPreviewFilter from "components/PostPreviewFilter";
+
+interface IBlogSectionHeader {
+  title?: string;
+  categoryFilter?: boolean;
+  categories?: string[];
+}
+
+const BlogSectionHeader = ({
+  title,
+  categoryFilter,
+  categories,
+}: IBlogSectionHeader) => {
+  return (
+    <>
+      <div className="blog-section-header">
+        <h2 className="title">{title}</h2>
+        {categoryFilter && categories ? (
+          <PostPreviewFilter options={categories} />
+        ) : null}
```

Comment:
❗ Improvement

Here, your component is getting to know too much of what's happening outside of it. You need to pass the `categoryFilter` and the `categories` variables just to render a filter conditionally. This plays against the composition pattern promoted in React.

Could you change this so that the one in control of the filter is the one component that needs it?